event-publisher/src/event_handler.rs

```diff
@@ -1,22 +1,37 @@
+use std::marker::PhantomData;
+
 use crate::{
     common::constants::MAX_BUFFER_MSGS, event_layer::EventLayer, publisher::MbusPublisher,
 };
 use events_api::{event::EventMessage, event_traits::initilize_source_component};
 
 /// Event handle.
-pub struct EventHandle {}
+pub struct EventHandle<T> {
+    __phantom: PhantomData<T>,
+}
 
-impl EventHandle {
+impl<T> EventHandle<T>
+where
```

Comment:
You don't need to add T for the struct btw, I think can just add it to the init function